src/sentry/static/sentry/app/components/events/interfaces/breadcrumbsV2/breadcrumbs.tsx

```diff
@@ -175,38 +201,56 @@ class BreadcrumbsContainer extends React.Component<Props, State> {
     };
   };
 
-  handleFilter = (breadcrumbFilterGroups: BreadcrumbFilterGroups) => () => {
+  handleFilter = (filterGroups: BreadcrumbFilterGroups) => () => {
     //types
-    const breadcrumbFilterGroupTypes = breadcrumbFilterGroups.filter(
-      breadcrumbFilterGroup => breadcrumbFilterGroup.groupType === 'type'
-    );
-
-    // TODO(Priscila): implement levels
-    // const breadcrumbFilterGroupLevels = breadcrumbFilterGroups
-    //   .filter(breadcrumbFilterGroup => breadcrumbFilterGroup.groupType === 'level')
-    //   .map(breadcrumbFilterGroup => breadcrumbFilterGroup.type);
+    const breadcrumbFilterGroupTypes = filterGroups
+      .filter(
+        breadcrumbFilterGroup =>
+          breadcrumbFilterGroup.groupType === 'type' && breadcrumbFilterGroup.isChecked
+      )
+      .map(breadcrumbFilterGroup => breadcrumbFilterGroup.type);
+
+    //levels
+    const breadcrumbFilterGroupLevels = filterGroups
+      .filter(
+        breadcrumbFilterGroup =>
+          breadcrumbFilterGroup.groupType === 'level' && breadcrumbFilterGroup.isChecked
+      )
+      .map(breadcrumbFilterGroup => breadcrumbFilterGroup.type);
+
+    this.setState(
+      prevState => ({
+        filteredByFilter: prevState.breadcrumbs.filter(({type, level}) => {
+          if (
+            breadcrumbFilterGroupLevels.length > 0 &&
+            breadcrumbFilterGroupTypes.length > 0
+          ) {
+            return (
+              breadcrumbFilterGroupTypes.includes(type) ||
+              (defined(level) && breadcrumbFilterGroupLevels.includes(level))
+            );
+          }
 
-    this.setState({
-      filteredBreadcrumbs: this.state.breadcrumbs.filter(breadcrumb => {
-        const foundBreadcrumbFilterData = breadcrumbFilterGroupTypes.find(
-          crumbFilterData => crumbFilterData.type === breadcrumb.type
-        );
-        if (foundBreadcrumbFilterData) {
-          return foundBreadcrumbFilterData.isChecked;
-        }
+          if (breadcrumbFilterGroupLevels.length > 0) {
+            return defined(level) && breadcrumbFilterGroupLevels.includes(level);
+          }
 
-        return false;
+          return breadcrumbFilterGroupTypes.includes(type);
+        }),
+        filterGroups,
       }),
-      breadcrumbFilterGroups,
-    });
+      () => {
+        this.handleFilterBySearchTerm(this.state.searchTerm);
+      }
+    );
   };
```

Comment:
Should the array filter be extracted into the named function? It feels messy to have that big of a code block inside setState.